sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -153,6 +153,12 @@ class CSVOptions(
   val samplingRatio =
     parameters.get("samplingRatio").map(_.toDouble).getOrElse(1.0)
 
+  /**
+   * Forcibly apply the specified or inferred schema to datasource files.
+   * If the option is enabled, headers of CSV files will be ignored.
+   */
+  val enforceSchema = getBool("enforceSchema", true)
```

Comment:
migration guide for behaviour change

Reply:
It doesn't change existing behavior. By default, user's or inferred schema is forcibly applied, and CSV headers are ignored completely.

Reply:
Sorry I missied:

```
+          checkHeaderFlag = !parser.options.enforceSchema, caseSensitive)
```

Can't we simply name the option `checkHeader`?

Reply:
`enforceSchema` is from Reynold. I am also fine about the name. This name can be applied to the other data sources later.

Reply:
As @gatorsmile wrote the name was selected in the perspective that similar option can be applied to another datasources that contains type info explicitly. CSV doesn't types. How would you propose to check types?

Reply:
I didn't know 'schema' implies source's schema. Yup, if there's a plan for it, I'm okay. we can talk about this later too for the naming.